cli/command_repository_create.go

```diff
@@ -49,7 +49,7 @@ func (c *commandRepositoryCreate) setup(svc advancedAppServices, parent commandP
 	cmd.Flag("object-splitter", "The splitter to use for new objects in the repository").Default(splitter.DefaultAlgorithm).EnumVar(&c.createSplitter, splitter.SupportedAlgorithms()...)
 	cmd.Flag("create-only", "Create repository, but don't connect to it.").Short('c').BoolVar(&c.createOnly)
 	cmd.Flag("format-version", "Force a particular repository format version (1, 2 or 3, 0==default)").IntVar(&c.createFormatVersion)
-	cmd.Flag("retention-mode", "Set the blob retention-mode for supported storage backends.").EnumVar(&c.retentionMode, blob.Governance.String(), blob.Compliance.String())
+	cmd.Flag("retention-mode", "Set the blob retention-mode for supported storage backends.").EnumVar(&c.retentionMode, blob.Governance.String(), blob.Compliance.String(), blob.Locked.String())
```

Comment:
While Kopia works with multiple tiers underneath, but the storage APIs and CLIs are designed to work seamlessly across all with minimal storage specific couplings. There are a few places where it is just unavoidable to expose storage specific parameters to the CLI/API.

Maybe we should rename the retention-modes in Kopia from "governance" / "compliance" to something more generic like "WORM Mode" / "RBAC Mode" or similar to keep things generic between S3/GCS/Azure/etc. However, this can be done in a separate PR; no need to overload this PR.

I am wondering whether we need to make an exception for Azure immutability as well. For all intents and purposes of Azure usage within Kopia, ain't Azure "locked" mode is just the same as the governance mode ? So, for Azure can we use the existing mode of Kopia called "governance" as its locked mode ? That way, we'll not need to introduce a new mode and going forward we can make things more generic and rename governance to "WORM Mode" or something more generic so it doesn't sound like AWSish.

Reply:
Locked is more like Compliance mode (Governance is not so secure as certain users can override the protections).
The main reasons against using Compliance:
1. Azure is kind of like a 2nd class citizen then
2. In the `azure_storage` we'd need to replace Compliance with Locked in put & extend methods
Can be done but looks a bit strange imo - if GCP eventually adds support then they may also have different names at that point too

"WORM Mode" / "RBAC Mode" I guess would be a breaking change for existing setups

Reply:
I've updated it to use COMPLIANCE although I didn't update https://github.com/kopia/kopia/pull/3405 - it could be updated when this is merged to make use of the `blob.Locked` from this PR

Reply:
Thanks Mike. You have valid points. I am not hard on it, but thanks for making the change. Next time when we make the change of "WORM Mode" / "RBAC Mode" then we'll have to do less number of these.

Reply:
I'm leaning more to going back to Locked...it also need adding to `putBlob` too which I forgot about and that's part of the problem of using S3 names where they don't belong...you need to remember to update them everywhere

Reply:
No worries, thanks for trying.

Reply:
What is the ugly IF check BTW ?

Reply:
We should try to provide an uniform interface in the CLI. We only had a single immutability implementation (for S3) and unfortunately we did not anticipate creating that interface.

Here's a straw-man proposal, please beat it and find holes.

What about accepting all these (case-insensitive) values in the CLI for both S3 and Azure.
1. "Locked"
2. "Compliance"
3. "Unlocked"
4. "Governance"

And having an internal translation to a canonical `{locked, unlocked}` representation.
And then in each provider translate from the canonical representation to the provider specific setting.

For S3 object locking:
- `locked` => `"Compliance"`
- `unlocked` => `"Governance"`

For Azure immutability:
- `locked` => `"Locked"`
- `unlocked` => `"Unlocked"`

Thoughts?

Again, let's do this in a separate PR, and merge this one after making the requested changes.